### PR TITLE
Fix KeyError in ComponentContext

### PR DESCRIPTION
**Pull request by TanguyMorelle**
## About this pull request

Fix a KeyError raised when deselecting all choices of a Select with multiple choices

## Changes

Set the `selected_options` attribute to and empty list when the `value` key is missing

## Checklist

- [x] I've run the `pre_push.py` script to format and lint code.
- [x] I've checked this pull request runs on `Python 3.6.X`.
- [ ] This fixes something in [Issues](https://github.com/eunwoo1104/discord-py-slash-command/issues).
    - Issue:
- [ ] This adds something new.
- [ ] There is/are breaking change(s).
- [ ] (If required) Relevant documentation has been updated/added.
- [ ] This is not a code change. (README, docs, etc.)

### CODE REVIEW

**Comment by LordOfPolls** [comment]
Hey, can you run `pre-push.py` as right now this code violates our coding style.

Besides that, good catch, waiting to merge

**Comment by TanguyMorelle** [comment]
Should be good now !

**Comment by LordOfPolls** [comment]
Why'd you force push?

**Comment by TanguyMorelle** [comment]
Just to amend the commit, you'd rather I make a second one ? (I squashed the modification commit into the first one)

File: discord_slash/context.py [hunk 1]
```diff
@@ -321,7 +321,12 @@ def __init__(
             )
             self.component = self.origin_message.get_component(self.custom_id)
 
-        self.selected_options = _json["data"]["values"] if self.component_type == 3 else None
+        self.selected_options = None
+
+        if self.component_type == 3:
+            self.selected_options = (
+                _json["data"]["values"] if "values" in _json["data"].keys() else []
```
Comment (benwoo1110) [inline discord_slash/context.py]: `.keys()` is not needed?

Comment (LilSpazJoekp) [reply discord_slash/context.py]: Correct. Doing `item in dictionary` same as `item in dictionary.keys()`.

File: discord_slash/context.py [hunk 2]
```diff
@@ -321,7 +321,12 @@ def __init__(
...
+            )
```
Comment (LilSpazJoekp) [inline discord_slash/context.py]:
```suggestion
            self.selected_options = _json["data"].get("values", [])
```
This can be simplified.

**Review by LilSpazJoekp — CHANGES_REQUESTED** [review]

File: discord_slash/context.py (hunk 1 above)
Comment (LordOfPolls) [reply discord_slash/context.py]: This has been resolved

**Comment by LordOfPolls** [comment]
Thanks for contributing, merging

**Pull request merged**